system/lib/wasmfs/wasmfs.cpp

```diff
@@ -119,4 +128,57 @@ __wasi_errno_t __wasi_fd_read(
   *nread = num;
   return __WASI_ERRNO_INVAL;
 }
+
+__wasi_fd_t __syscall_open(long pathname, long flags, long mode) {
+  int accessMode = (flags & O_ACCMODE);
+  bool canWrite = false;
+
+  if (accessMode == O_WRONLY || accessMode == O_RDWR) {
+    canWrite = true;
+  }
+
+  std::string interim = "";
+  std::vector<std::string> pathParts;
+
+  auto newPathName = (const char*)pathname;
+
+  for (int i = 0; newPathName[i] != '\0'; i++) {
+    if (newPathName[i] != '/') {
+      interim += newPathName[i];
+    } else {
+      // Avoid adding empty string
+      if (i > 0) {
+        pathParts.push_back(interim);
+      }
+      interim = "";
+    }
+  }
+  // Case where there is no / at the end of the string.
+  pathParts.push_back(interim);
```

Comment:
Might be worth looking in `src/library_path.js` to check for any special cases in the file path splitting logic.

(Also, this will need to be a utility function eventually, like it is there, since it will have multiple callers, but can leave that for later.)